app/controllers/photo-albums/photo-album/photos/index.js

```diff
@@ -8,15 +8,38 @@ export default Controller.extend(FileSaverMixin, {
   fetch: service(),
   photoSorting: ['-createdAt'],
   sortedPhotos: sort('model.photos', 'photoSorting'),
+  modalIsOpen: false,
+  downloadStarted: false,
+  downloadFinished: false,
+  modalTitle: 'Fotoalbum downloaden',
+  modalText: 'Het inpakken en downloaden van het album kan even duren. Terwijl de download bezig is kun je de webstek niet gebruiken en dit tabblad niet sluiten.',
   actions: {
     downloadAlbum() {
       let model = this.get('model');
+      this.set('downloadStarted', true);
+      this.set('modalTitle', 'Dit kan even duren...');
```

Comment:
Kan je niet deze text vast in het template zetten, ipv dynamisch op deze manier doen? Dus gewoon een {{#if ...}} Op basis van downloadStarted oid?

Misschien een veld introduceren `downloadStatus` met notStarted, pending, success en error?

Reply:
Als dat netter is kan het wel, ik dacht dat het juist mooier was om niet te veel ifs in de templates te stoppen, en dan houd je de functionaliteit overzichtelijker. Nu is er namelijk maar een plek waar alle waardes gezet worden en is het duidelijker te zien welke gevolgen er zijn.

Reply:
De reden dat ik het netter vind is dat er dan zo min mogelijk code in de routes zit, dan hoef ik niet in twee bestanden tegelijk op te zoeken wat er precies gebeurt: ik heb graag een zo voorspelbaar mogelijke route/controller/component

Reply:
Check, doe ik dat.